src/snake.cpp

```diff
@@ -417,19 +417,31 @@ void snake::snake::getInput()
             {
                 case KEY_UP:
                 case 'w':
-                    input = moveUp;
+                    if (!snake::snake::noOppDir)
+                        input = moveUp;
+                    if (input != moveDown)
+                        input = moveUp;
                     break;
                 case KEY_DOWN:
                 case 's':
-                    input = moveDown;
+                    if (!snake::snake::noOppDir)
```

Comment:
namespace and class identifiers are not needed, if the function is inside the same class / namespace. You can just do !noOppdir